test/user.test.js

```diff
@@ -1763,6 +1763,54 @@ describe('User', function() {
     });
   });
 
+  describe('password reset with/without email verification', function() {
+    it('allows resetPassword by email if email verification is required and done',
+  function(done) {
+    User.settings.emailVerificationRequired = true;
+    var email = 'foo1@bar.com';
+    var calledBack = false;
+
+    User.resetPassword({ email: 'foo1@bar.com' }, function() {
+      calledBack = true;
+    });
+
+    User.once('resetPasswordRequest', function(info) {
+      assert(info.email);
+      assert(info.accessToken);
+      assert(info.user.emailVerified);
+      done();
+    });
+  });
+
+    it('disallows resetPassword by email if email verification is required and not done',
+  function(done) {
+    User.settings.emailVerificationRequired = true;
+    var email = 'foo@bar.com';
```

Comment:
This is difficult to understand. Please use `validCredentialsEmail.email` instead.